### PR TITLE
chore: [IOPLT-603] Sets Android API Level 34 

### PR DESCRIPTION
**Pull request by CrisTofani**
> [!note]
> Depends on #5864 

## Short description
This PR sets Android API Level to 34 and upgrades `react-native-vision-camera` to solve compatibility issues

## List of changes proposed in this pull request
- Sets Android API Level to 34
- Removes `vision-camera-code-scanner`
- Updates `react-native-vision-camera`
- Refactors the app components using the camera to scan Barcode

## How to test
Test barcode code scanning to check everything is ok on both iOS and Android. Normally use the App on Android to check if the new API Level bump compromises anything

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -6,8 +6,8 @@ buildscript {
         buildToolsVersion = "33.0.2"
```
Comment (Vangaorth) [inline android/build.gradle:6]: Should be also switch to `buildTollsVersion` to `34.0.0`?

**Review by Vangaorth — CHANGES_REQUESTED** [review]

**Review by Vangaorth — APPROVED** [review]
Build tools updated to 34.0.0
react-native-share to latest version (in order to support share on Android 14)

Tested on each message flow and also the sharing of the payment receipt.

LGTM

**Review by shadowsheep1 — APPROVED** [review]
LGTM: Tested 
- Build
- EIC Login
- Camera Permissions on both Android 14 and 12.

We need just to remember that for the future:

```shell
[VisionCamera] react-native-worklets-core not found, Frame Processors disabled!
WARNING:We recommend using a newer Android Gradle plugin to use compileSdk = 34

This Android Gradle plugin (7.4.2) was tested up to compileSdk = 33

This warning can be suppressed by adding
    android.suppressUnsupportedCompileSdk=34
to this project's gradle.properties

The build will continue, but you are strongly encouraged to update your project to
use a newer Android Gradle Plugin that has been tested with compileSdk = 34
```

**Review by LazyAfternoons — APPROVED** [review]
LGTM, tested the QR code scan on the following devices without noticing any regression: 
- iPhone XS@17;
- Samsung S23@14;
- Pixel 4a@13;
- Honor 9@9;
- Samsung A51@13.

**Pull request merged**